homeassistant/components/fan/__init__.py

```diff
@@ -45,6 +45,7 @@
 SPEED_OFF = 'off'
 SPEED_LOW = 'low'
 SPEED_MEDIUM = 'medium'
+SPEED_MED = 'med'
```

Comment:
This should not be added to the main fan file. Fans use SPEED_MEDIUM. Neither will it ever be passed into `set_speed` from the frontend.

I think what @Teagan42 mentioned is that the states had to do with discovery. You will have to see why they are still not getting discovered even after the fix has been applied.